dragonfly/block/grass_plant.go

```diff
@@ -0,0 +1,157 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/dragonfly/block/grass"
+	"github.com/df-mc/dragonfly/dragonfly/item"
+	"github.com/df-mc/dragonfly/dragonfly/item/tool"
+	"github.com/df-mc/dragonfly/dragonfly/world"
+	"github.com/go-gl/mathgl/mgl64"
+	"math/rand"
+)
+
+// GrassPlant is a transparent plant block which can be used to obtain seeds and as decoration.
+type GrassPlant struct {
+	noNBT
+	transparent
+	empty
+
+	UpperBit bool
+
+	grass.Grass
+}
+
+// FlammabilityInfo ...
+func (g GrassPlant) FlammabilityInfo() FlammabilityInfo {
+	if g.Grass == grass.NetherSprouts() {
+		return FlammabilityInfo{
+			Encouragement: 60,
+			Flammability:  0,
+			LavaFlammable: true,
+		}
+	}
+	return FlammabilityInfo{
+		Encouragement: 60,
+		Flammability:  100,
+		LavaFlammable: true,
+	}
+}
+
+// BreakInfo ...
+func (g GrassPlant) BreakInfo() BreakInfo {
+	return BreakInfo{
+		Hardness:    0,
+		Harvestable: alwaysHarvestable,
+		Effective:   nothingEffective,
+		Drops: func(t tool.Tool) []item.Stack {
+			if g.Grass == grass.NetherSprouts() {
+				return []item.Stack{item.NewStack(g, 1)}
+			}
+			if rand.Float32() > 0.57 {
+				return []item.Stack{item.NewStack(WheatSeeds{}, 1)}
+			}
+			return []item.Stack{}
+		},
+	}
+}
+
+// BoneMeal attempts to affect the block using a bone meal item.
+func (g GrassPlant) BoneMeal(pos world.BlockPos, w *world.World) bool {
+	switch g.Grass {
+	case grass.SmallGrass():
+		w.SetBlock(pos, GrassPlant{Grass: grass.TallGrass()})
+		w.SetBlock(pos.Side(world.FaceUp), GrassPlant{Grass: grass.TallGrass(), UpperBit: true})
+		return true
+	case grass.Fern():
+		w.SetBlock(pos, GrassPlant{Grass: grass.LargeFern()})
+		w.SetBlock(pos.Side(world.FaceUp), GrassPlant{Grass: grass.LargeFern(), UpperBit: true})
```

Comment:
Are these supposed to trigger block updates around it? If so this should use PlaceBlock.

Reply:
It isn't, which is why I use SetBlock.